util/daisy_board_gen.py

```diff
@@ -0,0 +1,294 @@
+		]\
+	},\
+	Encoder: {\
+		typename: "daisy::Encoder",\
+		pin: "a,b,click",\
+		process: "${name}.Debounce();",\
+		updaterate: "${name}.SetUpdateRate(seed.AudioCallbackRate());",\
+		mapping: [\
+			{\
+				name: "${name}",\
+				get: "(hardware.${name}.Increment()*0.5f+0.5f)",\
+				range: [-1, 1]\
+			},\
+			{\
+				name: "${name}_press",\
+				get: "(hardware.${name}.Pressed()?1.f:0.f)",\
+				range: [0, 1]\
+			},\
+			{\
+				name: "${name}_rise",\
+				get: "(hardware.${name}.RisingEdge()?1.f:0.f)",\
+				range: [0, 1]\
+			},\
+			{\
+				name: "${name}_fall",\
+				get: "(hardware.${name}.FallingEdge()?1.f:0.f)",\
+				range: [0, 1]\
+			},\
+			{\
+				name: "${name}_seconds",\
+				get: "(hardware.${name}.TimeHeldMs()*0.001f)",\
+				range: null\
+			}\
+		]\
+	},\
+	GateIn: {\
+		typename: "daisy::GateIn",\
+		pin: "a",\
+		mapping: [\
+			{ name: "${name}", get: "(hardware.${name}.State()?1.f:0.f)", range: [0, 1] },\
+			{ name: "${name}_trig", get: "(hardware.${name}.Trig()?1.f:0.f)", range: [0, 1] }\
+		]\
+	},\
+	AnalogControl: {\
+		typename: "daisy::AnalogControl",\
+		pin: "a",\
+		flip: false,\
+		invert: false,\
+		slew: "1.0/seed.AudioCallbackRate()",\
+		process: "${name}.Process();",\
+		updaterate: "${name}.SetSampleRate(seed.AudioCallbackRate());",\
+		mapping: [{ name: "${name}", get: "(hardware.${name}.Value())", range: [0, 1] }]\
+	},\
+	Led: {\
+		typename: "daisy::Led",\
+		pin: "a",\
+		invert: true,\
+		postprocess: "${name}.Update();",\
+		mapping: [{ name: "${name}", set: "hardware.${name}.Set($<name>);" }]\
+	},\
+	RgbLed: {\
+		typename: "daisy::RgbLed",\
+		pin: "r,g,b",\
+		invert: true,\
+		postprocess: "${name}.Update();",\
+		mapping: [\
+			{ name: "${name}_red", set: "hardware.${name}.SetRed($<name>);" },\
+			{ name: "${name}_green", set: "hardware.${name}.SetGreen($<name>);" },\
+			{ name: "${name}_blue", set: "hardware.${name}.SetBlue($<name>);" },\
+			{ name: "${name}", set: "hardware.${name}.Set(clamp(-$<name>, 0.f, 1.f), 0.f, clamp($<name>, 0.f, 1.f));" },\
+			{ name: "${name}_white", set: "hardware.${name}.Set($<name>,$<name>,$<name>);" }\
+		]\
+	},\
+	GateOut: {\
+		typename: "daisy::dsy_gpio",\
+		pin: "a",\
+		mode: "DSY_GPIO_MODE_OUTPUT_PP",\
+		pull: "DSY_GPIO_NOPULL",\
+		mapping: [\
+			{ name: "${name}", set: "dsy_gpio_write(&hardware.${name}, $<name> } 0.f);" }\
+		]\
+	},\
+	CVOuts: {\
+		typename: "daisy::DacHandle::Config",\
+		pin: "",\
+		bitdepth: "daisy::DacHandle::BitDepth::BITS_12",\
+		buff_state: "daisy::DacHandle::BufferState::ENABLED",\
+		mode: "daisy::DacHandle::Mode::POLLING",\
+		channel: "daisy::DacHandle::Channel::BOTH",\
+		mapping: [\
+			{\
+				name: "${name}1",\
+				set: "hardware.${name}.WriteValue(daisy::DacHandle::Channel::ONE, $<name> * 4095)",\
+				where: "main"\
+			},\
+			{\
+				name: "${name}2",\
+				set: "hardware.${name}.WriteValue(daisy::DacHandle::Channel::TWO, $<name> * 4095)",\
+				where: "main"\
+			}\
+		]\
+	}\
+}'
```

Comment:
is this defined as plane text within the script in the oopsy version?

It might be nice to just keep this in a file, `component_defaults.json` or something? 

Also not sure if this was local to oopsy or not either, but there seems to be mixed use of the template tags, though that could totally be intentional (i.e. `${name}` and `$<name>`).

Reply:
It was defined as a javascript object, whichlooks the same as JSON for obvious reasons.
I'll move it to a json file rather than having a giant string hanging around in the pyhton.

Reply:
Cool! Sounds good.

And yeah, I'm actually looking over some other oopsy stuff right now and it seems like there is a difference:
`$<name>` is the value of the parameter, where `${name}` is a template substitution for the name of the parameter